src/_pytest/unittest.py

```diff
@@ -275,23 +276,20 @@ def addExpectedFailure(
     def addUnexpectedSuccess(
         self, testcase: "unittest.TestCase", reason: str = ""
     ) -> None:
-        self._store[unexpectedsuccess_key] = reason
+        msg = "Unexpected success"
+        if reason:
+            msg += f": {reason}"
+        strict = self.config.getini("xfail_strict")
+        self._store[xfailed_key] = Xfail(
+            reason=msg, run=True, strict=strict, raises=None
+        )
 
     def addSuccess(self, testcase: "unittest.TestCase") -> None:
         pass
 
     def stopTest(self, testcase: "unittest.TestCase") -> None:
         pass
 
-    def _expecting_failure(self, test_method) -> bool:
```

Comment:
perhaps we could leverage `_expecting_failure()` and apply pytest xfal mark on each of them?

Reply:
Can you explain to what you refer by "them"? (Don't quite follow the question).

Also, IIUC, `_expecting_failure` is currently (in master) just dead code right?

Reply:
> _expecting_failure is currently (in master) just dead code right?

Yes.

>Can you explain to what you refer by "them"? 

"them" -> test methods with expectedFailure/todo mark. 
Somethig like - `if _expecting_failure(test_method):` -> `test_method.add_mark(pytest.mark.xfail)`. But now I think it would rather complicate things.